Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -5815,19 +5818,21 @@ class BBQJIT {
                 emitStoreConst(Value::fromF64(0), m_locals[i]);
                 break;
             case TypeKind::I64:
-            case TypeKind::Ref:
-            case TypeKind::Structref:
             case TypeKind::Struct:
             case TypeKind::Rec:
-            case TypeKind::Funcref:
-            case TypeKind::Externref:
             case TypeKind::Func:
             case TypeKind::Array:
-            case TypeKind::Arrayref:
             case TypeKind::Sub:
-            case TypeKind::RefNull:
                 emitStoreConst(Value::fromI64(0), m_locals[i]);
                 break;
+            case TypeKind::Externref:
+            case TypeKind::Funcref:
+            case TypeKind::Ref:
+            case TypeKind::RefNull:
+            case TypeKind::Structref:
+            case TypeKind::Arrayref:
+                emitStoreConst(Value::fromI64(bitwise_cast<uint64_t>(JSValue::encode(jsNull()))), m_locals[i]);
```

Comment:
We already have an `encodedJSUndefined()`.  I think the fact that we're doing this encoding and casting so often suggest that we should add an `encodedJSNull()`.

Reply:
I feel `JSValue::encode(jsNull())` is more clear in terms of reading. This kind of form is used in DFG / FTL etc. everywhere, and it looks very obvious what it returns.